ComponentKitTests/TransactionalDataSource/CKTransactionalComponentDataSourceIntegrationTests.mm

```diff
@@ -0,0 +1,123 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <XCTest/XCTest.h>
+#import <OCMock/OCMock.h>
+
+#import "CKComponent.h"
+#import "CKComponentProvider.h"
+#import "CKComponentScope.h"
+#import "CKCompositeComponent.h"
+#import "CKComponentController.h"
+#import "CKCollectionViewTransactionalDataSource.h"
+#import "CKTransactionalComponentDataSourceConfiguration.h"
+#import "CKTransactionalComponentDataSourceChangeset.h"
+
+static NSNotificationCenter *_notificationCenter = nil;
+
+@interface CKTransactionalComponentDataSourceIntegrationTestComponent : CKCompositeComponent
+@end
+
+@implementation CKTransactionalComponentDataSourceIntegrationTestComponent
++ (instancetype)newWithIdentifier:(id)identifier {
+  CKComponentScope scope(self, identifier);
+  return [self newWithComponent:[CKComponent new]];
+}
+@end
+
+@interface CKTransactionalComponentDataSourceIntegrationTestComponentController : CKComponentController
+@end
+
+@implementation CKTransactionalComponentDataSourceIntegrationTestComponentController
+- (void)didUpdateComponent {
+  [super didUpdateComponent];
+  [_notificationCenter postNotificationName:NSStringFromSelector(_cmd) object:self];
+}
+- (void)willRemount {
+  [super willRemount];
+  [_notificationCenter postNotificationName:NSStringFromSelector(_cmd) object:self];
+}
+- (void)didRemount {
+  [super didRemount];
+  [_notificationCenter postNotificationName:NSStringFromSelector(_cmd) object:self];
```

Comment:
Oh for sure! I'm open to ideas for a rewrite. The "problem" is that, since the controllers are entirely uncoupled, there's no more direct way to access them. This is kind of just the first thing that came to mind...